Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -2778,3 +2778,37 @@ Function Set-AVSVSANClusterUNMAPTRIM {
         }
     }
 }
+
+<#
+    .DESCRIPTION
+     Set default Multipath Policy to Round Robin.
+
+    .PARAMETER DatastoreName
+     Datastore name
+
+    .EXAMPLE
+     Set-DatastoreMultipathingPolicy -DatastoreName "myDatastore"
+
+    .INPUTS
+     vCenter datastore name.
+
+    .OUTPUTS
+     None.
+#>
+function Set-DatastoreMultipathingPolicy {
```

Comment:
This seems to be specific to NMP plugin and not HPP. Maybe the name of the method should refer to it ? 
Other datastores, such as NVMe/TCP uses HPP and the round robin policy is set via the HPP plugin and not NMP those this method will not take effect.

For which datastore transports you want to control the multipath policy here ?

Note you are setting the policy to a specific value "VMW_PSP_RR".  Shouldn't the method receive as a parameter the desired policy and then apply it ?  (VMW_PSP_MRU or VMW_PSP_FIXED)

Reply:
1. Yes, its specific to NMP plugin for now, we are not implementing it for HPP, we are not targeting NVMe currently.
2.  I’m sorry, but I don’t understand your question. Could you please rephrase it?
3.  For now all Pure customers have agreed on Round Robin Policy unanimously therefore we want to implement it as default policy.

Reply:
For further clarification regarding the script kindly contact - [tyagi.prashant@microsoft.com](mailto:tyagi.prashant@microsoft.com), [mchhibber@microsoft.com](mailto:mchhibber@microsoft.com), we can get in call as well as this feature is required to be implemented on urgent basis.

Reply:
I agree with @abel-lb we should try and keep the cmdlets generic. Since this will be invoked by automation, perhaps the params can be hard-coded on the calling side instead of the implementation?

Reply:
@Malika133 ^ this I'd still like to see answered, did we consider making the script generic and pinning the correct values on the caller side?
Considerations: What happens if we choose to change the default? As this script version will forever be available the caller will be pinned to this policy, unless we change the script _and_ they change the client to invoke the new version. 
It's not wrong, but why do we need to change 2 things in sync if it could be controlled entirely on the client?

Reply:
Yes Eugene, we have made the script generic, it will accept the Multipathing Policy as parameter by the user while creating VMFS datastore, in case user does not provide any Policy the default policy will be implemented.

Reply:
I have removed default selection now Eugene, if user doesn't not select any value default by VMware for VMFS will be implemented.